app/src/org/commcare/activities/ReminderThread.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * 
+ */
+package org.commcare.activities;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.Vector;
+
+import org.commcare.CommCareApplication;
+import org.commcare.suite.model.Alert;
+import org.javarosa.core.model.condition.EvaluationContext;
+import org.javarosa.core.model.instance.DataInstance;
+import org.javarosa.core.model.instance.ExternalDataInstance;
+import org.javarosa.core.model.instance.TreeReference;
+import org.javarosa.model.xform.XPathReference;
+import org.javarosa.xpath.XPathParseTool;
+import org.javarosa.xpath.expr.XPathExpression;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.media.Ringtone;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.util.Log;
+
+/**
+ * @author ctsims, Saumya Jain
+ *
+ */
+public class ReminderThread {
+    Runnable runnable;
+    Thread myThread;
+    
+    HashMap<String, String> keyToValue;
+    HashMap<String, String> valueToKey;
+
+    private String caseTypeString = "";
+    private String xPathConditionString;
+
+    Context mContext;
+    
+    Cursor c;
+    
+    private boolean mContinuePolling = true;
+    private int sleepTime = 10;
+    
+    public ReminderThread(Context context) {
+        valueToKey = new HashMap<>();
+        keyToValue = new HashMap<>();
+        
+        this.mContext = context;
+        
+        runnable = new Runnable() {
+            @Override
+            public void run() {
+
+                //Prepare xpath expression
+                XPathExpression condition;
+                try{
+                    condition = XPathParseTool.parseXPath(xPathConditionString);
+                }catch (Exception e){
+                    e.printStackTrace();
+                    return;
+                }
+
+                //Prepare evaluation context
+                Hashtable<String, DataInstance> instances = new Hashtable<>();
+                instances.put("casedb", new ExternalDataInstance("jr://instance/casedb","casedb"));
+                EvaluationContext ec = CommCareApplication._().getCurrentSessionWrapper().getEvaluationContext(instances);
+
+                //Prepare TreeReferences
+                String caseXPathRef = "instance('casedb')/casedb/case[@case_type='" + caseTypeString + "']";
```

Comment:
Rather than loading the caseType as the external element here, I think we want to shift the Alert to load this full string (the "nodeset" reference), which would give us more flexibility in setting what types of reminders are available